test/integration/model/create.test.js

```diff
@@ -931,7 +931,26 @@ describe(Support.getTestDialectTeaser('Model'), () => {
     });
 
     if (dialect === 'postgres') {
-      it('does not cast arrays for postgresql insert', function() {
+      it('create arrays for postgresql', function() {
+        const User = this.sequelize.define('UserWithArray', {
+          myvals: { type: Sequelize.ARRAY(Sequelize.INTEGER) },
+          mystr: { type: Sequelize.ARRAY(Sequelize.STRING) }
+        });
+
+        let test = false;
+        return User.sync({ force: true }).then(() => {
+          return User.create({ myvals: [1, 2, 3], mystr: ['One', 'Two'] }, {
+            logging(sql) {
+              test = true;
+              expect(sql).to.contain('INSERT INTO "UserWithArrays" ("id","myvals","mystr","createdAt","updatedAt") VALUES (DEFAULT,ARRAY[1,2,3],ARRAY[\'One\',\'Two\']');
```

Comment:
Here you are checking an incomplete SQL (abruptly ended at `ARRAY[\'One\',\'Two\']`), I think this could cause confusion. Please assert on the complete SQL. I suggest updating the `User` model you create in this test to not have timestamps (`{ timestamps: false }`) so that you don't have to go over the hassle of asserting the `createdAt`/`updatedAt` fields.

Reply:
Good idea!